index.js

```diff
@@ -0,0 +1,75 @@
+const https = require('https');
+const { parseString } = require('xml2js');
+
+async function getAllUrlsFromSitemapIndex(siteUrl) {
+  try {
+    const sitemapIndexUrl = `${siteUrl}/sitemap.xml`;
+    const sitemapIndexResponse = await fetchContent(sitemapIndexUrl);
+
+    const sitemapIndexXml = sitemapIndexResponse.data;
+    const sitemapIndexResult = await parseXml(sitemapIndexXml);
+    const sitemapUrls = sitemapIndexResult.sitemapindex.sitemap.map(sitemap => sitemap.loc[0]);
+    const fetchSitemapPromises = sitemapUrls.map(url => fetchContent(url));
+    const sitemapResponses = await Promise.all(fetchSitemapPromises);
+
+    let allUrls = [];
+    for (const response of sitemapResponses) {
+      const sitemapResult = await parseXml(response.data);
+
+      if (sitemapResult.urlset && sitemapResult.urlset.url) {
+        const sitemapUrls = sitemapResult.urlset.url.map(urlObj => urlObj.loc[0]);
```

Comment:
Not sure but, It will be better if we call db insertion call here because each URL will be found  here. So, In getAllUrlsFromSitemapIndex function again we doesn't need to iterate the loop for those many times.